server_modules/scores.js

```diff
@@ -44,7 +71,54 @@ exports.route = function(app) {
                 return score.published && score.round === round;
             });
             res.json(scoresForRound);
-        }).catch(utils.sendError(res)).done();
+        }).catch(err => utils.sendError(res, err)).done();
+    });
+
+    //save a new score
+    app.post('/scores/create',function(req,res) {
+        var body = JSON.parse(req.body);
+        var scoresheet = body.scoresheet;
+        var score = body.score;
+
+        fileSystem.writeFile(fileSystem.getDataFilePath("scoresheets/" + score.file), req.body)
+        .then(changeScores(function(result) {
+            result.scores.push(score);
```

Comment:
You're right. Since I'm currently also working on the authentication and protection feature, I didn't think of the two-side-validation that might be needed here. The score is, of course, being sanitized by the client. Do you think there should be a validation step here? I will simply throw an error if the score doesn't match the sanitized interface

Reply:
Yes, I think there should be validation on the server too.

It's a small thing to add, and will save a lot of time debugging if it fails in the future (e.g. someone submitting a score using an older cached version in the browser).

Reply:
Will dp